Code/Framework/AzCore/AzCore/Debug/Trace.cpp

```diff
@@ -84,10 +83,18 @@ namespace AZ::Debug
     static constexpr auto ErrorEventId = EventNameHash("Error");
     static constexpr auto AssertEventId = EventNameHash("Assert");
 
-    constexpr LogLevel DefaultLogLevel = LogLevel::Info;
+    static void TraceLevelChanged(const int& newLevel)
+    {
+        Debug::ITrace::Instance().SetLogLevel(static_cast<LogLevel>(newLevel));
+    }
+
+    static void AlwaysShowCallstackChanged(const bool& enable)
+    {
+        Debug::ITrace::Instance().SetAlwaysPrintCallstack(enable);
+    }
 
-    AZ_CVAR_SCOPED(int, bg_traceLogLevel, DefaultLogLevel, nullptr, ConsoleFunctorFlags::Null, "Enable trace message logging in release mode.  0=disabled, 1=errors, 2=warnings, 3=info.");
-    AZ_CVAR_SCOPED(bool, bg_alwaysShowCallstack, false, nullptr, ConsoleFunctorFlags::Null, "Force stack trace output without allowing ebus interception.");
+    AZ_CVAR_SCOPED(int, bg_traceLogLevel, LogLevel::Info, &TraceLevelChanged, ConsoleFunctorFlags::Null, "Enable trace message logging in release mode.  0=disabled, 1=errors, 2=warnings, 3=info.");
+    AZ_CVAR_SCOPED(bool, bg_alwaysShowCallstack, &AlwaysShowCallstackChanged, nullptr, ConsoleFunctorFlags::Null, "Force stack trace output without allowing ebus interception.");
```

Comment:
The value parameter was replace with the callback function which is not going to work.
```suggestion
    AZ_CVAR_SCOPED(bool, bg_alwaysShowCallstack, false, &AlwaysShowCallstackChanged, ConsoleFunctorFlags::Null, "Force stack trace output without allowing ebus interception.");
```

Reply:
Gah! Good catch.